src/core_plugins/kibana/public/dashboard/dashboard.js

```diff
@@ -226,6 +240,43 @@ app.directive('dashboardApp', function ($injector) {
         $scope.indexPatterns = dashboardState.getPanelIndexPatterns();
       };
 
+      $scope.filter = function (field, value, operator, index) {
+        if ($scope.model.query.language === 'lucene') {
+          filterManager.add(field, value, operator, index);
+        }
+
+        if ($scope.model.query.language === 'kuery') {
+          const fieldName = _.isObject(field) ? field.name : field;
+          const kueryAST = fromKueryExpression($scope.model.query.query);
+          const newAST = addNode(
+            kueryAST,
+            nodeTypes.match.buildNode({ field: fieldName, values: value, operation: operator })
+          );
+          $scope.updateQuery({ query: toKueryExpression(newAST), language: 'kuery' });
+        }
+      };
+
+
+      $scope.$watch('model.query', (newQuery) => {
+        $scope.model.query = migrateLegacyQuery(newQuery);
+        dashboardState.applyFilters($scope.model.query, filterBar.getFilters());
+        $scope.refresh();
+      });
+
+      $scope.$watch(() => dashboardState.getAppState().$newFilters, function (filters = []) {
```

Comment:
Yeah, this is currently just replicating what the `filter_bar` already does for filters. My plan is to get rid of the `$newFilters` $scope variable entirely and change the `filter_bar_click_handler` so that it either takes a callback, or can just be used as a pure function. I actually have a branch with some of those changes already, but I re-focused on developing the query language itself since this is technical debt that we can clean up post 6.0 if necessary.